src/passes/Print.cpp

```diff
@@ -2106,6 +2106,25 @@ struct PrintSExpression : public OverriddenVisitor<PrintSExpression> {
       doIndent(o, indent);
       o << ";; custom section \"" << section.name << "\", size "
         << section.data.size();
+      bool isPrintable = true;
+      for (auto c : section.data) {
+        if (!isprint(c)) {
+          isPrintable = false;
+          break;
+        }
+      }
+      if (isPrintable) {
```

Comment:
The majority of custom sections are going to have `name` elements that include length fields that aren't really printable, so this won't generalize far beyond the sourceMappingURL section. Since many custom sections are tool conventions, though, I wonder if it would make sense to add code here to print each of them with knowledge of their layout and semantics. If that seems useful, perhaps this PR can just do the printing if the name of the section is "sourceMappingURL" instead of inferring if the contents are printable.

Reply:
`sourceMappingURL` was not the initial goal of this PR. It was introduced for testing in #2327, and `sourceMappingURL` just also happened to be printable when this code was added.

We could hardcode both names, but given this "lucky accident", I'd say let's ship the generic code as-is - 1) it's pretty low-cost since binary sections would quickly bail out, and 2) `--print` is used only for debugging anyway where such minor slowness wouldn't be noticeable and 3) it will make it work out of the box with other potential tooling.

Reply:
To be clear, I'm not concerned with performance; I'm concerned that we could write different code that would be more useful. I think properly printing custom sections would be very useful, but I see only limited value in a solution that bails out for most sections with no clear upgrade path.

I may be wrong that most custom sections are not directly printable, though. Besides `sourceMappingURL`, do you know which ones will be printed by this code?

Reply:
I do agree that most (aside from `sourceMappingURL` and `asyncify`) are not directly printable, but I don't see how it's worse than the current status quo. Basically, this PR at least gives a way to debug printable sections, and leaves the rest up to the implementer, like they were left before. I'm not going for the general implementation here, just fixing one thing at a time and trying to find a way to at least debug sections we already know how to print.

Reply:
This does seem reasonable to me. Even if it works for just one section, it's simple and helpful. We can always add custom section printing logics later.

One thing that does worry me though is what happens if the section is extremely large (but printable), like if it contains raw data that happens to be in the ascii range (like 10MB of `a`). Maybe there should be a limit?

Reply:
Heh yeah, I thought about that, but can't think of such an example section for now and it requires thinking about which limit would be reasonable :D If you have ideas (200 bytes? 1000? More?), I'm happy to add such limit. Or we can wait until someone runs into a real-world issue with this.

Reply:
Fair enough, can wait for a concrete issue I guess...